### DISCUSSION TITLE
Hyperlink break into multiple lines not render underline correctly

### DISCUSSION THREAD

**Discussion by kindy** [ghostty-org/ghostty#2780] [Q&A] [answered]
- Build: 8086
- Commit: [873ebc36](https://github.com/ghostty-org/ghostty/commits/873ebc36)

see this video for details:

https://github.com/user-attachments/assets/5873fb05-df2f-4498-9f01-7e624a384c2a

**Comment by kindy** [comment]
This seems more like a bug than a question?

**Reply by mitchellh** [reply]
Its not a bug.

**Comment by mitchellh** [comment, marked as answer]
This is not a bug. Vim is drawing the URL and is NOT soft-wrapping the line. So it is two lines, not one. Ghostty only unwraps lines that are wrapped. This is the equivalent to typing the following:

```
http://example
.com
```

Your browser wouldn't highlight that as one URL because its two lines. This is the same thing happening here.

The solution for programs (and things like neovim do) is to use OSC8 which is conceptually similar to wrapping something in anchor tags:

```
<a href="...">http://example
.com</a>
```

Now your browser would highlight the full URL. Terminals are the same (but use OSC8 instead of anchor tags). Ghostty supports OSC8, so I recommend using a vim configuration that also highlights using OSC8. At least in neovim, I find most filetype plugins do this.

**Comment by kindy** [comment]
I just test it, and seems vim insert `^[[3;1H` at end of the first line, instead of `\n`.

---

`cat -v ghostty-test.txt` 
```
a41
# See http://zsh.sourceforge.net/Doc/Release/Prompt-Expansion.html#Simple-Prompt-Escapes
z43
```

`stty -a | head -1`
```
speed 9600 baud; 18 rows; 58 columns;
```

`vim -R ghostty-test.txt > ooo-ghost.txt`  (quit with `:qa!<enter>`)

`cat -v ooo-ghost.txt`   (copy with |pbcopy)
```
^[[?1049h^[[>4;2m^[[?1h^[=^[[?2004h^[[?1004h^[[1;18r^[[?12h^[[?12l^[[22;2t^[[22;1t^[[27m^[[23m^[[29m^[[m^[[H^[[2J^[[?25l^[[18;1H"ghostty-test.txt" [readonly] 4L, 98B^[[1;1Ha41^M
# See http://zsh.sourceforge.net/Doc/Release/Prompt-Expanss^[[3;1Hion.html#Simple-Prompt-Escapes^M
z43^M
^M
^[[94m~                                                         ^[[7;1H~                                                         ^[[8;1H~                                                         ^[[9;1H~                                                         ^[[10;1H~                                                         ^[[11;1H~                                                         ^[[12;1H~                                                         ^[[13;1H~                                                         ^[[14;1H~                                                         ^[[15;1H~                                                         ^[[16;1H~                                                         ^[[17;1H~                                                         ^[[1;1H^[[?25h^[[?4m^[[?25l^[[m^[[18;1H^[[K^[[18;1H:qa^[[?25h!^M^[[?25l^[[?2004l^[[>4;m^[[23;2t^[[23;1t^[[18;1H^[[K^[[18;1H^[[?1004l^[[?2004l^[[?1l^[>^[[?1049l^[[?25h^[[>4;m
```

**Reply by 00-kat** [reply]
> I just test it, and seems vim insert `^[[3;1H` at end of the first line, instead of `\n`.

That's the escape sequence to move your cursor to row three, column one.  Either ways, it's not soft-wrapped so as far as Ghostty can tell they're two separate lines.

**Comment by kindy** [comment]
when running with glfw: `zig build -Dapp-runtime=glfw run`, underline of multiple line links can render sometime. like this:

https://github.com/user-attachments/assets/0ac297c4-a772-45d5-a9e2-0522a4d818dd
